packages/cli/lib/services/dryrun.service.ts

```diff
@@ -148,8 +149,23 @@ class DryRunService {
             messages: []
         };
 
+        // dry-run mode does not read or write to the records database
+        // so we can safely mock the records service
+        const recordsService: RecordsServiceInterface = {
+            markNonCurrentGenerationRecordsAsDeleted: async (
+                _connectionId: number,
+                _model: string,
+                _syncId: string,
+                _generation: number
+                // eslint-disable-next-line @typescript-eslint/require-await
+            ): Promise<string[]> => {
+                return [];
+            }
+        };
+
```

Comment:
In order to avoid having `shared` depends on `records` I have deciding to use dependency injection for the 2 cases where `records` must be accessed in `shared`.
One is in `run.service` to mark as deleted records when `track_delete = true`. The second one is in `sync.client` to delete all records when a FULL sync is triggered.

Here for dryrun, since `writeToDb` is false I am passing a mock records service, otherwise I construct one (see below in jobs)